OpenRA.Mods.Common/Activities/Air/FlyForward.cs

```diff
@@ -14,19 +14,32 @@
 
 namespace OpenRA.Mods.Common.Activities
 {
-	public class FlyTimed : Activity
+	public class FlyForward : Activity
 	{
 		readonly Aircraft aircraft;
 		readonly WDist cruiseAltitude;
-		int remainingTicks;
+		readonly int flyTicks;
+		int remainingDistance;
+		int ticks;
 
-		public FlyTimed(int ticks, Actor self)
+		FlyForward(Actor self)
 		{
-			remainingTicks = ticks;
 			aircraft = self.Trait<Aircraft>();
 			cruiseAltitude = aircraft.Info.CruiseAltitude;
 		}
 
+		public FlyForward(Actor self, int ticks)
```

Comment:
Style thought: Maybe define a default `ticks = -1` to make `FlyOffMap` a little cleaner?